x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanStreamOutput.java

```diff
@@ -225,10 +228,37 @@ public boolean writeEsFieldCacheHeader(EsField field) throws IOException {
             cacheId = cacheEsField(field);
             writeZLong(-1 - cacheId);
         }
-        writeString(field.getWriteableName());
+        writeCachedString(field.getWriteableName());
         return true;
     }
 
+    @Override
+    public void writeCachedString(String string) throws IOException {
+        if (getTransportVersion().onOrAfter(TransportVersions.ESQL_CACHED_STRING_SERIALIZATION)) {
```

Comment:
I think it'd be easier to read if we did `if getTransportVersion().before(ESQL_CACHED_STRING_SER) { writeString(); return;}`. I know it's less lines this way, but we call out the short case quickly then I don't have to think about it after reading the first line.